### ISSUE TITLE
Better way to organize torrents in the main view, e.g. by label or watch/download directory

### ISSUE THREAD

**Issue by josh-ua** [rakshasa/rtorrent#1256]
Hi Friends --

Is there anyway to better group torrents in the main view, based on a custom attribute set by the user? For example, only show torrents from a certain label or from within a certain watch/download directory  (set from the rtorrent config file).

Is this currently possible at all? Or is the main view just limited to all torrents and can only be filtered by the various states mentioned here? https://github.com/rakshasa/rtorrent/wiki/User-Guide#global-keys

Thanks

**Comment by stickz** [comment]
Yup, it's called a front-end for rTorrent. ruTorrent allows you to apply filters on top of filters. And automatically group torrents with expressions. It's mind boggling what can be done in PHP and JavaScript! https://github.com/Novik/ruTorrent

**Comment by josh-ua** [comment]
I'm aware of ruTorrent, but was looking for a solution that just utilized the base rtorrent. That doesn't exist, but I was able to replicate this functionality by running multiple instances of rtorrent. Of course, that doesn't scale well, but I only needed 2 or 3 degrees of separation and it's fine for me. For anyone wondering, here is a blog post that walks you through that set up: https://kernelwho.wordpress.com/2011/11/15/running-multiple-instances-of-rtorrent/

The tl;dr is make different rtorrent config files (make sure they are not named rtorrent.rc) with different sessions and port ranges and you can run as many rtorrent instances as your heart (and patience) desires.

**Comment by kannibalox** [comment]
For posterity, while it's not possible to change the main view, it is possible to create new views and filter them as needed:

```ini
view.add = movies
view.persistent = movies
method.insert =	d.is_movie, simple, "equal={d.custom1=,((cat,movies))}"
view.filter = movies, ((d.is_movie))
schedule2 = filter_movies, 22, 33, "view.filter = movies, d.is_movie="
```

Alternately the [pyrosimple tools](https://github.com/kannibalox/pyrosimple) (which I maintain, for full disclosure), make on-the-fly filtering very easy:

```bash
rtcontrol label=movies --to-view=movies
```

**Comment by josh-ua** [comment]
Thanks @kannibalox . I think that is more in line was I was looking for -- I was wrong, it does exist!  Will also take a look at your tools too.

**Comment by darkalchemy** [comment]
@kannibalox How would bind the new view to a key?
